.setup/bin/sample_courses/models/__init__.py

```diff
@@ -0,0 +1,264 @@
+            self.preferred_familyname = user["user_preferred_familyname"]
+        if "user_email" in user:
+            self.email = user["user_email"]
+        if "user_group" in user:
+            self.group = user["user_group"]
+        if self.group < 1 or 4 < self.group:
+            raise SystemExit(
+                "ASSERT: user {}, user_group is not between 1 - 4. Check YML file.".format(
+                    self.id
+                )
+            )
+        if "user_access_level" in user:
+            self.access_level = user["user_access_level"]
+        if self.access_level < 1 or 3 < self.access_level:
+            raise SystemExit(
+                "ASSERT: user {}, user_access_level is not between 1 - 3. Check YML file.".format(
+                    self.id
+                )
+            )
+        if "registration_section" in user:
+            self.registration_section = int(user["registration_section"])
+        if "rotating_section" in user:
+            self.rotating_section = int(user["rotating_section"])
+        if "grading_registration_section" in user:
+            self.grading_registration_section = user["grading_registration_section"]
+        if "unix_groups" in user:
+            self.unix_groups = user["unix_groups"]
+        if "manual_registration" in user:
+            self.manual = user["manual_registration"] is True
+        if "courses" in user:
+            self.courses = {}
+            if isinstance(user["courses"], list):
+                for course in user["courses"]:
+                    self.courses[course] = {"user_group": self.group}
+            elif isinstance(user["courses"], dict):
+                self.courses = user["courses"]
+                for course in self.courses:
+                    if "user_group" not in self.courses[course]:
+                        self.courses[course]["user_group"] = self.group
+            else:
+                raise ValueError(
+                    "Invalid type for courses key, it should either be list or dict"
+                )
+        if "sudo" in user:
+            self.sudo = user["sudo"] is True
+        if "user_password" in user:
+            self.password = user["user_password"]
+
+    def create(self, force_ssh=False):
+        if not DB_ONLY and not user_exists(self.id):
+            if self.group > 2 and not force_ssh:
+                self.create_non_ssh()
+            else:
+                self.create_ssh()
+            self.create_ldap()
+
+        if self.group <= 1:
+            add_to_group("submitty_course_builders", self.id)
+        if self.sudo:
+            add_to_group("sudo", self.id)
+
+    def create_ssh(self):
+        print("Creating user {}...".format(self.id))
+        os.system(
+            "useradd -m -c 'First Last,RoomNumber,WorkPhone,HomePhone' {}".format(
+                self.id
+            )
+        )
+        self.set_password()
+
+    def create_non_ssh(self):
+        print("Creating user {}...".format(self.id))
```

Comment:
Would using f-strings like in line 170 be better than .format(variable) in making the files consistent? 

This could also be in a future PR...

Reply:
Yep, I will change it in the next PR, hopefully after this PR gets merged